lib/api/CInferenceModelDefinition.cc

```diff
@@ -377,48 +487,43 @@ CTrainedModel::TStringVec CTree::removeUnusedFeatures() {
 }
 
 std::string CInferenceModelDefinition::jsonString() const {
+    std::ostringstream jsonStrm;
+    this->jsonStream(jsonStrm);
+    return jsonStrm.str();
+}
 
-    std::ostringstream stream;
-    {
-        core::CJsonOutputStreamWrapper wrapper{stream};
-        CSerializableToJson::TRapidJsonWriter writer{wrapper};
-        rapidjson::Value doc{writer.makeObject()};
-        this->addToDocument(doc, writer);
-        writer.write(doc);
-        stream.flush();
-    }
-    // string writer puts the json object in an array, so we strip the external brackets
-    std::string jsonStr{stream.str()};
-    std::string resultString(jsonStr, 1, jsonStr.size() - 2);
-    return resultString;
+void CInferenceModelDefinition::jsonStream(std::ostream& jsonStrm) const {
+    rapidjson::OStreamWrapper wrapper{jsonStrm};
+    CSerializableToJson::TGenericLineWriter writer{wrapper};
+    this->addToJsonStream(writer);
+    jsonStrm.flush();
 }
 
-std::stringstream CInferenceModelDefinition::jsonStringCompressedFormat() const {
+std::stringstream CInferenceModelDefinition::jsonCompressedStream() const {
     std::stringstream compressedStream;
     using TFilteredOutput = boost::iostreams::filtering_stream<boost::iostreams::output>;
-    std::string modelDefinitionStr{jsonString()};
     {
         TFilteredOutput outFilter;
         outFilter.push(boost::iostreams::gzip_compressor());
         outFilter.push(core::CBase64Encoder());
         outFilter.push(compressedStream);
-        outFilter << modelDefinitionStr;
+        this->jsonStream(outFilter);
         outFilter.flush();
```

Comment:
I would have thought it would automatically flush on destruction.

Reply:
Indeed, removed it now.